library/std/src/sync/mpmc/mod.rs

```diff
@@ -360,9 +360,18 @@ impl<T> Sender<T> {
     /// that a return value of [`Err`] means that the data will never be
     /// received, but a return value of [`Ok`] does *not* mean that the data
     /// will be received. It is possible for the corresponding receiver to
-    /// hang up immediately after this function returns [`Ok`].
+    /// hang up immediately after this function returns [`Ok`]. If the channel
+    /// is zero-capacity, however, the channel becomes a rendezvous channel and
+    /// it guarantees that the receiver has indeed received the data if this
+    /// function returns success.
```

Comment:
This wording is a bit tricky. Maybe something like "However, if the channel is zero-capacity, it acts as a rendezvous channel and an `Ok(())` result indicates the data was successfully received" or similar.

Otherwise LGTM.

Reply:
Thanks, looks good with one nit: could you change "is successfully received" to "has been" or "was"?

After that please squash then r=me.

Reply:
Done.